controllers/utils/dns_test.go

```diff
@@ -185,12 +204,31 @@ func TestValidEdgeDNSButNonExistingFQDN(t *testing.T) {
 	edgeDNSServer := "localhost"
 	fqdn := "some-valid-ip-fqdn-123"
 	// act
-	result, err := Dig(fqdn, DNSServer{Host: edgeDNSServer, Port: 53})
+	result, err := Dig(fqdn, 8, DNSServer{Host: edgeDNSServer, Port: 53})
 	// assert
 	assert.Error(t, err)
 	assert.Nil(t, result)
 }
 
+func TestDigCNAME(t *testing.T) {
+	testServer := DNSServer{
+		Host: server,
+		Port: port,
+	}
+	NewFakeDNS(testSettings).
+		AddCNAMERecord("foo.cloud.example.com.", "bar.cloud.example.com.").
+		AddARecord("bar.cloud.example.com.", net.IPv4(10, 1, 0, 3)).
```

Comment:
Can we add one more CNAME here and set `maxRecursion` to 1. Just to check if the decrement is working